### ISSUE TITLE
Add support for MultipartFile

### ISSUE THREAD

**Issue by Hronom** [spring-cloud/spring-cloud-openfeign#62]
I have tested `spring-cloud-openfeign` of version `2.0.1` and it not automatically process `MultipartFile`.

This means that constructions like this:
```
@FeignClient(value = "file-service")
public interface FileServiceClient {
    @PostMapping(value = "/upload-file", consumes = MediaType.MULTIPART_FORM_DATA_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    ResponseEntity<String> processFileTest(
        @RequestPart(value = "file", required = true) MultipartFile file
    );
}
```
**not works.**

To make it work - you need.
**1.**
Include in your `pom.xml` dependency:
```
        <dependency>
            <groupId>io.github.openfeign.form</groupId>
            <artifactId>feign-form-spring</artifactId>
            <version>3.3.0</version>
        </dependency>
```

**2.**
Configure you feign client with next configuration:
```
@Configuration
public class ServiceClientConfig {
    @Autowired
    private ObjectFactory<HttpMessageConverters> messageConverters;

    @Bean
    public Encoder feignFormEncoder() {
        return new SpringFormEncoder(new SpringEncoder(messageConverters));
    }
}
```

Only after that, you will be able to use `@RequestPart(value = "file", required = true) MultipartFile file,`.
This should be supported in Spring Boot and Cloud by default. 

As far as I understand this change should be added only here org.springframework.cloud.openfeign.FeignClientsConfiguration or somewhere else?

Related discussion was started here https://github.com/spring-cloud/spring-cloud-netflix/issues/867#issuecomment-416911321

**Comment by antemooo** [comment]
**TL;DR** <br />**Transform your MultiPartFile to MultiValueMap. [See example that i provide on StackOverFlow ](https://stackoverflow.com/a/58641199/9159029)**

----------
The answer mentioned by [@Hronom](https://github.com/Hronom) is correct answer to have your FeignClient method signature the same as the actual endpoint signature that you are trying to call. However, there is a way around that does not require you to define a FormEncoder configuration and add any extra dependency because in some applications you are not allowed to do that (enterprise shit); all what you need is to transform your MultipartFile to a MultiValueMap and it will perfectly work as the standard encoder will be able to serialize it.
